File: Python/Chapter03/MyStack.py
import sys

class MyStack:
	'''push-down スタッククラス'''
	
	def __init__(self, stackSize):
		'''コンストラクタ（初期化）'''
		
		# フィールド
		self.myStack = [ ' ' for i in range(stackSize) ]
		self.top  = -1;
		
	def pushDown(self, data):
		'''プッシュダウン'''
		
		self.top += 1
		if (self.top > len(self.myStack)):
			print('stack overflow', file=sys.stderr)
			sys.exit(0)
			
		self.myStack[self.top] = data
		
	def popUp(self):
		'''ポップアップ'''
		
		if (self.top < 0):
			print('popup from empty stack', file=sys.stderr)
			sys.exit(0)
		
		self.top -= 1
		
		return self.myStack[self.top+1]
		
	def isEmpty(self):
		'''空か否か'''
		
		if (self.top < 0):
			return True
		else:
			return False

File: Java/Chapter10/MyStack.java
public class MyStack {

    /* フィールド */
    char[] myStack;
    int top;

    /* コンストラクタ（初期化）*/
    public MyStack(int stackSize){
	myStack = new char[stackSize];
	top = -1;
    }

    public void pushDown(char data){
	top++;
	if (top >= myStack.length) {
	    System.err.println("stack overflow");
	    System.exit(0);
	}
	myStack[top] = data;
    }	

    public char popUp(){
	if (top < 0) {
	    System.err.println("popup from empty stack");
	    System.exit(0);
	}
	top--;
	return myStack[top+1];
    }

    public boolean isEmpty(){
	if (top < 0)
	    return true;
	else
	    return false;
    }


}
